typescript/infra/src/config/agent/agent.ts

```diff
@@ -158,17 +159,30 @@ export abstract class AgentConfigHelper<R = unknown>
 {
   connectionType: AgentConnectionType;
   docker: DockerConfig;
+  chainDockerOverrides?: Record<ChainName, Partial<DockerConfig>>;
```

Comment:
Tried to make it slightly more clear by renaming the helper fn to `dockerImageForChain`. I'm not super worried here because this is mostly a stopgap until we merge the Solana feature branch and can use the same image for all validators. We also don't read the docker image in many places & if there's misuse of it in the future, the effect of it shouldn't be too bad

Reply:
I'd prefer to not have it be per-chain because as we expand the # of chains I worry it can get unwieldy

Reply:
kinda feel the same
would prefer general `chain => agentconfig` overrides instead of individual value overrides if that makes sense

Reply:
I think that'd only work for validators (bc that's the only kind of agent that has distinct helm releases per chain), and would come with a pretty large surface area of changes

I agree with both your sentiments that there's room for better configuration to accommodate changes throughout this PR but imo that's a scope increase I don't have the time / team buy in for. My goal with these was to get something working with the intent of revisiting later when we choose to make changes throughout to better accommodate future alt execution environments, so I'm hesitant to spend more time changing a lot of things for a smaller payoff that may be refactored away in the future